src/components/calcite-date-picker/calcite-date-picker.tsx

```diff
@@ -0,0 +1,495 @@
+import {
+  Component,
+  h,
+  Prop,
+  Event,
+  Element,
+  Host,
+  State,
+  Listen,
+  Build,
+  EventEmitter,
+  Watch,
+  VNode
+} from "@stencil/core";
+import { getLocaleData, DateLocaleData } from "./utils";
+import { getElementDir } from "../../utils/dom";
+import { dateFromRange, dateFromISO, dateToISO, getDaysDiff } from "../../utils/date";
+
+import { getKey } from "../../utils/key";
+import { TEXT } from "./calcite-date-picker-resources";
+
+import { DateRangeChange } from "../../interfaces/DateRangeChange";
+
+@Component({
+  assetsDirs: ["calcite-date-picker-nls"],
+  tag: "calcite-date-picker",
+  styleUrl: "calcite-date-picker.scss",
+  shadow: true
+})
+export class CalciteDatePicker {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteDatePickerElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+  /** Selected date */
+  @Prop() value?: string;
+
+  /** Selected date as full date object*/
+  @Prop({ mutable: true }) valueAsDate?: Date;
```

Comment:
I'm not sure why some of these props are mutable but it seems like we should be able to get away from that.